### PR TITLE
Fixed issue with multiple polling queries

### PR DESCRIPTION
**Pull request by Poincare**
Since we were using a single timer reference, there were polling leaks when we had multiple polling queries - see #248 

This changeset should fix that issue and introduces a unit test that used to fail but now doesn't.

TODO:

- [x] Update CHANGELOG.md with your change
- [x] Make sure all of the significant new logic is covered by tests
- [x] Rebase your changes on master so that they can be merged easily
- [x] Make sure all tests and linter rules pass

### CODE REVIEW

File: src/QueryManager.ts [hunk 1]
```diff
@@ -94,7 +94,7 @@ export class QueryManager {
   private networkInterface: NetworkInterface;
   private store: ApolloStore;
   private reduxRootKey: string;
-  private pollingTimer: NodeJS.Timer | any; // oddity in typescript
+  private pollingTimers: {[queryId: string]: NodeJS.Timer | any};
```
Comment (helfer) [inline src/QueryManager.ts]: Remind me, why does this have to have an `any`? If it's an oddity in typescript, maybe we should keep that explanation.

File: test/QueryManager.ts [hunk 2]
```diff
@@ -1471,6 +1471,121 @@ describe('QueryManager', () => {
...
+      }
+    );
+    const queryManager = new QueryManager({
+      networkInterface,
+      store: createApolloStore(),
+      reduxRootKey: 'apollo',
+    });
+    const handle1 = queryManager.watchQuery({
+      query: query1,
+      pollInterval: 800,
```
Comment (helfer) [inline test/QueryManager.ts]: Did you try setting these to smaller numbers so the test can run in less time? 1.6 sec seems like a long time. I'm guessing the smaller pollInterval could be cut to 50ms, maybe even less.

**Comment by helfer** [comment]
This is great! Just wondering if we can cut the time necessary for the test.

**Comment by jbaxleyiii** [comment]
@Poincare this is a great improvement

File: src/QueryManager.ts (hunk 1 above)
Comment (Poincare) [inline src/QueryManager.ts]: I don't really know why. I will find out and add in and explanation.

File: test/QueryManager.ts (hunk 2 above)
Comment (Poincare) [inline test/QueryManager.ts]: The bigger numbers (or, bigger differences, at least) make sure that the test won't go haywire even if the timing is a bit screwy. But, I agree they can be made smaller so I'll go ahead and do that.

**Comment by helfer** [comment]
@Poincare Can you rebase this? As soon as it's rebased, we can merge it, and then rebase & merge #251

Awesome to see so many great PRs this week!

**Comment by Poincare** [comment]
Rebased.

**Pull request merged**